src/NuGet.Core/NuGet.Commands/PackagesLockFileBuilder.cs

```diff
@@ -89,5 +94,56 @@ public PackagesLockFile CreateNuGetLockFile(LockFile assetsFile)
             return lockFile;
         }
 
+        public async Task<PackagesLockFile> CreateNuGetLockFileAsync(IEnumerable<PackageReference> installedPackages, Func<PackageIdentity, string> getNupkgFilePath, CancellationToken token)
+        {
+            Debug.Assert(installedPackages != null);
+            Debug.Assert(getNupkgFilePath != null);
+
+            var lockFile = new PackagesLockFile();
+
+            foreach (var targetFramework in installedPackages.GroupBy(p => p.TargetFramework).OrderBy(g => g.Key))
+            {
+                var target = new PackagesLockFileTarget
+                {
+                    TargetFramework = targetFramework.Key,
+                    Dependencies = new List<LockFileDependency>()
+                };
+
+                foreach (var package in targetFramework.OrderBy(p => p.PackageIdentity))
+                {
+                    string hash;
+
+                    var file = getNupkgFilePath(package.PackageIdentity);
+                    if (file == null)
+                    {
+                        throw new Exception("couldn't find package");
+                    }
+
+                    using (var reader = new PackageArchiveReader(file))
+                    {
+                        hash = reader.GetContentHashForSignedPackage(token);
+                        if (hash == null)
+                        {
+                            hash = Convert.ToBase64String(await reader.GetArchiveHashAsync(HashAlgorithmName.SHA512, token));
+                        }
+                    }
+
+                    var dependency = new LockFileDependency
+                    {
+                        Id = package.PackageIdentity.Id,
+                        Type = PackageDependencyType.Direct,
+                        RequestedVersion = package.AllowedVersions,
```

Comment:
In packages.config world, `RequestedVersion` and `ResolvedVersion` would point to same metadata `packageIdentity.Version`

Reply:
I decided to remove the line, since RequestedVersion won't be read during packages.config restore.

Reply:
For the sake of consistency, I'd say you keep both requestedVersion as well as resolvedVersion in the lock file so that tomorrow anyone crawling this lock file gets consistent behavior for PC or PR.